spec/features/add_an_ip_address_spec.rb

```diff
@@ -24,16 +24,31 @@
 
     context 'and that IP is valid' do
       before do
+        ENV['S3_PUBLISHED_LOCATIONS_IPS_BUCKET'] = 'StubBucket'
+        ENV['S3_PUBLISHED_LOCATIONS_IPS_OBJECT_KEY'] = "StubKey"
+
+        allow_any_instance_of(Gateways::S3).to receive(:upload)
+      end
+
+      it 'calls locations_ips publishing' do
+        expect_any_instance_of(PublishLocationsIps).to receive(:execute)
```

Comment:
I wouldn't add this sort of mock to an acceptance spec - you're checking the internals of the system rather than external behaviour, which will make these tests brittle.

If I wanted to test this in an acceptance spec, I'd try to get as close to the external behaviour of the system as possible with what I'm checking.

Checking something is in a bucket seems impractical, so I'd probably check the right endpoints were called with the right data (I'm guessing adding something to a bucket is an API call across HTTP?).

Reply:
That is a positive move, but not sure if changing it is worth it (feels like we end up with a spec that's closer to the edge, but still suffers from the same brittleness as before).

This might be a little tough to do "correctly" if the gem is hooked-in system-wide, don't mind leaving it as is for speed's sake as it sounds like some work to sort out.